helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -573,35 +573,41 @@ public void resetPartition(String clusterName, String instanceName, String resou
     // check the instance is alive
     LiveInstance liveInstance = accessor.getProperty(keyBuilder.liveInstance(instanceName));
     if (liveInstance == null) {
-      throw new HelixException(
-          "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-              + ", because " + instanceName + " is not alive");
+      // check if the instance has never exist in the cluster
+      String instancePath = PropertyPathBuilder.instance(clusterName, instanceName);
+      String errMessage;
+      if (!_zkClient.exists(instancePath)) {
+        errMessage = String.format(
+            "Can't reset state for %s.%s on %s, because %s has never existed" + " in cluster %s",
+            resourceName, partitionNames, instanceName, instanceName, clusterName);
+      } else {
+        errMessage = String.format("Can't reset state for %s.%s on %s, because %s does not live in"
+                + " cluster %s anymore", resourceName, partitionNames, instanceName, instanceName,
+            clusterName);
+      }
+      throw new HelixException(errMessage);
     }
 
     // check resource group exists
     IdealState idealState = accessor.getProperty(keyBuilder.idealStates(resourceName));
     if (idealState == null) {
-      throw new HelixException(
-          "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-              + ", because " + resourceName + " is not added");
+      throw new HelixException(String
+          .format("Can't reset state for %s.%s on %s, because %s is not added", resourceName,
+              partitionNames, instanceName, resourceName));
     }
 
     // check partition exists in resource group
     Set<String> resetPartitionNames = new HashSet<String>(partitionNames);
+    Set<String> partitions;
     if (idealState.getRebalanceMode() == RebalanceMode.CUSTOMIZED) {
-      Set<String> partitions = new HashSet<String>(idealState.getRecord().getMapFields().keySet());
-      if (!partitions.containsAll(resetPartitionNames)) {
-        throw new HelixException(
-            "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-                + ", because not all " + partitionNames + " exist");
-      }
+      partitions = idealState.getRecord().getMapFields().keySet();
     } else {
-      Set<String> partitions = new HashSet<String>(idealState.getRecord().getListFields().keySet());
-      if (!partitions.containsAll(resetPartitionNames)) {
-        throw new HelixException(
-            "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-                + ", because not all " + partitionNames + " exist");
-      }
+      partitions = idealState.getRecord().getListFields().keySet();
+    }
+    if (!partitions.containsAll(resetPartitionNames)) {
+      throw new HelixException(String
+          .format("Can't reset state for %s.%s on %s, because not all %s exist", resourceName,
```

Comment:
It looks like there are many error messages look like this, I think we can have a private method to generate the formated messaged based on the input, the parameters could be: partitionNames, instanceName, failure reason (so the callers just have different logic of generating the different reasons).